### ISSUE TITLE
Blaze Client: Timeout results in unhelpful error

### ISSUE THREAD

**Issue by runarorama** [http4s/http4s#305]
When `PooledHttp1Client` connections time out (given the timeout in the `PooledHttp1Client` constructor), the following exception is thrown:

```
java.nio.channels.ClosedChannelException
at org.http4s.client.blaze.Http1ClientStage$$anonfun$org$http4s$client$blaze$Http1ClientStage$$executeRequest$1$$anonfun$apply$1.apply(Http1ClientStage.scala:87)
```

It would be super helpful to get an exception that indicates a timeout before the NIO hits the unexpected EOF.

**Comment by bryce-anderson** [comment]
Thanks for the ticket. Could you specify the blaze version you're using? I'm going to guess 0.8.1 but let me know if its something else.

**Comment by rossabaker** [comment]
"Bug" label was a bit harsh, but still, :+1: to improving this experience.

**Comment by bryce-anderson** [comment]
This should be fixed in this specific case by commit 1f1e1606f2f290b152b2f172. @runarorama, thanks for the bug report and please reopen if need be.
